avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1178,36 +1278,30 @@ private Schema toConnectSchema(org.apache.avro.Schema schema, boolean forceOptio
       case ENUM:
         // enums are unwrapped to strings and the original enum is not preserved
         builder = SchemaBuilder.string();
+        builder.parameter(CONNECT_ENUM_DOC_PROP, schema.getDoc());
+        builder.parameter(AVRO_TYPE_ENUM, schema.getFullName());
+        for(String enumSymbol : schema.getEnumSymbols()) {
+          builder.parameter(AVRO_TYPE_ENUM+"."+enumSymbol, enumSymbol);
+        }
         break;
 
       case UNION: {
-        // Connect doesn't support unions. To handle this, we convert them to records with field
-        // names associated with each of the input types
-        //
-        // However, we also need to first check for union types that are used to indicate optional
-        // fields. Since we don't support the null type, we'll consider any union containing a
-        // null type as an indication that it should be handled as an optional field
-        boolean hasNullSchema = false;
-        org.apache.avro.Schema optionalSchema = null;
-        for (org.apache.avro.Schema memberSchema : schema.getTypes()) {
-          if (memberSchema.getType() == org.apache.avro.Schema.Type.NULL) {
-            hasNullSchema = true;
-          } else {
-            optionalSchema = memberSchema;
+        if (schema.getTypes().size() == 2){
```

Comment:
Just to make sure we have compatibility sorted: we were already handling cases with null schemas that were really just simple optional schemas (i.e. 2 schemas). Since we were throwing exceptions otherwise, the only thing we've done here is to enable a previously disallowed set of schemas (unions with null and 2+ other schemas). Is that accurate?